File: cdsexamples/JavaExample/src/main/java/VMOptions.java
package cdsexamples;

import java.lang.management.ManagementFactory;
import java.lang.management.RuntimeMXBean;
import java.util.List;

class VMOptions {
    static List<String> get() {
        RuntimeMXBean runtimeMxBean = ManagementFactory.getRuntimeMXBean();
        return runtimeMxBean.getInputArguments();
    }
    private static final String sep = System.lineSeparator()+"   ";
    static String asString() {
        return "VM Options:"+sep+String.join(sep, VMOptions.get());
    }
}

File: cdsexamples/DottyExample/src/main/scala/VMOptions.scala
package cdsexamples

import java.lang.management.ManagementFactory
import java.lang.management.RuntimeMXBean

import scala.jdk.CollectionConverters._
import scala.language.implicitConversions

object VMOptions {

  def get: List[String] = {
    val runtimeMxBean = ManagementFactory.getRuntimeMXBean()
    runtimeMxBean.getInputArguments().asScala.toList
  }

  private val sep = System.lineSeparator()+"   "
  def asString: String = "VM Options:"+sep+String.join(sep, VMOptions.get: _*)

}
